File: ProjetoScania2/ProjetoScania2/Models/Vagas.cs
using System.ComponentModel.DataAnnotations;

namespace ProjetoVaga2.Models
{
    public class Vagas
    {
        public int id { get; set; }
        public int idempresa { get; set; }
        public int idturno { get; set; }
        public int idmodalidade { get; set; }
        public int idescolaridade { get; set; }
        public string? titulo_vaga { get; set; }
        public string? descricao { get; set; }
        public double salario { get; set; }
        public DateTime data_limite_inscricao { get; set; }
        public int quantidade_vaga { get; set; }
    }
}

File: app-projeto-04/src/app/classes/vagas.ts
export class Vagas {

    constructor() {
        this.id = 0;
        this.idempresa = 0;
        this.idturno = 0;
        this.idmodalidade = 0;
        this.idescolaridade = 0;
        this.titulo_vaga = '';
        this.descricao = '';
        this.salario = 0;
        this.data_limite_inscricao = '';
        this.quantidade_vaga = 0;
    }
    public id: number;
    public idempresa: number;
    public idturno: number;
    public idmodalidade: number;
    public idescolaridade: number;
    public titulo_vaga: string;
    public descricao: string;
    public salario: number;
    public data_limite_inscricao: string;
    public quantidade_vaga: number;
}
